### PR TITLE
move code generation, build and test to FAKE

### LINKED ISSUE #169: Replace msbuild tasks with Fake targets
This should make builds on mono work as well modernising the build chain.

### PR DESCRIPTION
**Pull request by kjnilsson**
Fixes #169 

    - try to get winrt build working
    - change appveyor config to use build.bat instead of msbuild
    - ensure build.fsx is committed
    - do not include apigen project in main build
    - check for appveyor environment variable
    - ensure winrt codegen is performed
    - copyright year update
    - clean up project files
    - remove assembly info from test project
    - fix assembly signing
    - appveyor config changes
    - dist.sh changes
    - turn appveyor auto tests off
    - make the handle leak tests a bit more reliable
    - remove Local.props file
    - make the Pack target work
    - amend test documentation and reduce handle leak test sleep
    - Add handle leak test to new category 'GCTest' and exclude this from appveyor build

### CODE REVIEW

**Comment by michaelklishin** [comment]
This will go into `3.6.3`/`3.6.x`. Please do not merge before `3.6.2` ships.

**Comment by michaelklishin** [comment]
I've tested building the client on Mono 4 and running tests on OS X. It works well: not only it works again, things build much quicker and the test runner provides progress output.

👍

File: RUNNING_TESTS.md [hunk 1]
```diff
@@ -12,24 +12,18 @@ It is possible to use [Visual Studio Community edition](https://www.visualstudio
 
 ## Building
 
-### Copy `Local.props.example` and Edit
+Simply run
 
-First copy `Local.props.example` to `Local.props` and edit it if needed, specifying whether
-you intend to build the client on Mono and what .NET version should be targeted.
+    build.sh|bat
```
Comment (michaelklishin) [inline RUNNING_TESTS.md]: I'd simply make it say `build.bat` and `When running on Mono, use build.sh` on the next line.

File: RUNNING_TESTS.md [hunk 2]
```diff
@@ -12,24 +12,18 @@ It is possible to use [Visual Studio Community edition](https://www.visualstudio
...
 ## Running Tests
 
 Tests can be run from Visual Studio using [NUnit Test Adapter](https://visualstudiogallery.msdn.microsoft.com/6ab922d0-21c0-4f06-ab5f-4ecd1fe7175d).
 Note that it may take some time for the adapter to discover tests in the assemblies.
 
-Running the tests from the command line is also straightforward: use
+Running the tests from the command line is also straightforward you don't need to build first to do this: use
 
-    msbuild.exe /t:RunUnitTests projects/client/Unit/RabbitMQ.Client.Unit.csproj
+    ./run-test.sh|bat
```
Comment (michaelklishin) [inline RUNNING_TESTS.md]: Same as above, I'd word it to make it possible to copy the command (and make it unambiguous to those new to cross-platform and/or *nix development).

**Comment by michaelklishin** [comment]
So we have 3 Mono-specific failures, one Mono bug and 2 "not implemented" exceptions in a test that is safe to skip on Mono:

```
Tests run: 213, Errors: 3, Failures: 0, Inconclusive: 0, Time: 671.3246077 seconds
  Not run: 0, Invalid: 0, Ignored: 0, Skipped: 0

Errors and Failures:
1) Test Error : RabbitMQ.Client.Unit.TestAmqpUri.TestAmqpUriParseSucceed
   System.UriFormatException : Invalid URI: The URI scheme is not valid.
  at System.Uri.Parse (UriKind kind, System.String uriString) [0x00000] in <filename unknown>:0
  at System.Uri.ParseUri (UriKind kind) [0x00000] in <filename unknown>:0
  at System.Uri..ctor (System.String uriString, UriKind uriKind) [0x00000] in <filename unknown>:0
  at RabbitMQ.Client.ConnectionFactory.set_Uri (System.String value) [0x00000] in <filename unknown>:0
  at RabbitMQ.Client.Unit.TestAmqpUri.ParseSuccess (System.String uri, System.String user, System.String password, System.String[] hosts, Int32 port, System.String vhost) [0x00000] in <filename unknown>:0
  at RabbitMQ.Client.Unit.TestAmqpUri.TestAmqpUriParseSucceed () [0x00000] in <filename unknown>:0
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0

2) Test Error : RabbitMQ.Client.Unit.TestConnectionChurnHandleLeak.TestHandleLeakWithDisabledHeartbeats
   System.NotImplementedException : The requested feature is not implemented.
  at System.GC.WaitForFullGCComplete (Int32 millisecondsTimeout) [0x00000] in <filename unknown>:0
  at RabbitMQ.Client.Unit.TestConnectionChurnHandleLeak.PerformLeakTest (RabbitMQ.Client.ConnectionFactory cf) [0x00000] in <filename unknown>:0
  at RabbitMQ.Client.Unit.TestConnectionChurnHandleLeak.TestHandleLeakWithDisabledHeartbeats () [0x00000] in <filename unknown>:0
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0

3) Test Error : RabbitMQ.Client.Unit.TestConnectionChurnHandleLeak.TestHandleLeakWithEnabledHeartbeats
   System.NotImplementedException : The requested feature is not implemented.
  at System.GC.WaitForFullGCComplete (Int32 millisecondsTimeout) [0x00000] in <filename unknown>:0
  at RabbitMQ.Client.Unit.TestConnectionChurnHandleLeak.PerformLeakTest (RabbitMQ.Client.ConnectionFactory cf) [0x00000] in <filename unknown>:0
  at RabbitMQ.Client.Unit.TestConnectionChurnHandleLeak.TestHandleLeakWithEnabledHeartbeats () [0x00000] in <filename unknown>:0
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0
```

@kjnilsson can we make those tests conditional on Mono? Ideally using NUnit's facilities but using an `if` in the actual tests would also be OK. While this is a bit out of scope for this PR, we are restoring Mono compatibility as far as developing the client goes, so this seems appropriate to me.

**Comment by kjnilsson** [comment]
ok @michaelklishin  I have addressed your comments.

**Pull request merged**